src/Security/Authentication/Twitter/src/TwitterHandler.cs

```diff
@@ -318,5 +326,79 @@ private string ComputeSignature(string consumerSecret, string tokenSecret, strin
                 return Convert.ToBase64String(hash);
             }
         }
+
+        private async Task EnsureTwitterRequestSuccess(HttpResponseMessage response)
+        {
+            var contentTypeIsJson = Equals(response.Content.Headers.ContentType, new MediaTypeHeaderValue("application/json"));
+            var contentTypeIsXml = Equals(response.Content.Headers.ContentType, new MediaTypeHeaderValue("application/xml"));
+
+            if (!response.IsSuccessStatusCode &&
+                (contentTypeIsJson || contentTypeIsXml))
+            {
+                TwitterErrorResponse errorResponse = null;
+                try
+                {
+                    // Failure, attempt to parse Twitters error message
+                    var errorContentStream = await response.Content.ReadAsStreamAsync();
+
+                    if (contentTypeIsJson)
+                    {
+                        errorResponse = await JsonSerializer.DeserializeAsync<TwitterErrorResponse>(errorContentStream, ErrorSerializerOptions);
+                    }
+                    else if (contentTypeIsXml)
+                    {
+                        // Load into XML doc and manually parse to same object
+                        var xmlDocument = new XmlDocument();
+                        xmlDocument.Load(errorContentStream);
+
+                        errorResponse = new TwitterErrorResponse
+                        {
+                            Errors = new List<TwitterError>()
+                        };
+
+                        var errors = xmlDocument.SelectSingleNode("//errors") ?? throw new XmlException("Error parsing Twitter XML error response");
```

Comment:
Style: Throwing exceptions just to catch them a few lines later is poor control flow and slow. Change this to an if condition.
```
if (errors != null)
{
  errorResponse = new TwitterErrorResponse
  {
    Errors = new List<TwitterError>()
  };
  foreach ...
}
```